app/controllers/comments_controller.rb

```diff
@@ -314,7 +314,15 @@ def permit_commenter
   def user_blocked?
     return false if current_user.blocked_by_count.zero?
 
-    UserBlock.blocking?(@comment.commentable.user_id, current_user.id) ||
-      UserBlock.blocking?(@comment.parent&.user_id, current_user.id)
+    blocked_by_commentable_author = UserBlock.blocking?(@comment.commentable.user_id, current_user.id)
+    blocked_by_comment_author = @comment.parent && UserBlock.blocking?(@comment.parent.user_id, current_user.id)
+
+    blocked_by_commentable_author || blocked_by_comment_author || blocker_upthread?(@comment.parent)
+  end
+
+  def blocker_upthread?(comment)
+    return false unless comment&.parent
+
+    UserBlock.blocking?(comment.parent.user_id, current_user.id) || blocker_upthread?(comment.parent)
```

Comment:
We can avoid recursion (harder to comprehend) and multiple queries with smth like:
```ruby
thread_authors_ids = comment.ancestors.pluck(:user_id) 
UserBlock.exists?(blocker_id: thread_authors_ids, blocked_id: blocked_id)
```
(the code needs to be checked for correctness)

Reply:
Thanks for this! I was kind of worried about how the recursion would play out both in terms of readability and performance